README.md
# Phonebook_Project_using_JAVA
 


History.txt
Washim Akram 01737861695


Mohammad Ali 01521520330
foysal ahmed 01993551641
Alien King 01737861696


src/phonebook/Edit.java

package phonebook;

import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.io.File;
import java.io.*;
import java.io.IOException;
import java.util.Scanner;
import javax.swing.*;

public class Edit extends JFrame implements ActionListener{
    String f_name,l_name,number,data,item[],name1="temp.txt",name="History.txt",check,dot2,copy;
    JButton button1=new JButton("Edit");
    JButton button2=new JButton("Delete");
    JButton button3=new JButton("Save");
    JPanel panel=new JPanel();
    JLabel label1=new JLabel("First Name");
    JTextField field1=new JTextField(20);
    JLabel label2=new JLabel("Last Name");
    JTextField field2=new JTextField(20);
    JLabel label3=new JLabel("Contact Number");
    JTextField field3=new JTextField(17);
    JFrame frame1=new JFrame();
    JFrame frame2=new JFrame();
    
    Edit(String data){
        super("Edit");
        this.data=data;
        //check save the old data field
        Scanner sc2=new Scanner(data);
                    while(sc2.hasNext()){
                        check=sc2.next();
                        break;
                    }
        JLabel label=new JLabel("       "+data);
        label.setFont(new Font("Times New Roman", 1, 24));
        
        setSize(550,240);
        add(panel);
        panel.add(label);
        
        label1.setFont(new Font("Times New Roman", 1, 18));
        field1.setFont(new Font("Times New Roman", 1, 18));
        
        label2.setFont(new Font("Times New Roman", 1, 18));
        field2.setFont(new Font("Times New Roman", 1, 18));
        
        label3.setFont(new Font("Times New Roman", 1, 18));
        field3.setFont(new Font("Times New Roman", 1, 18));
        
        panel.add(button1);
        panel.add(button2);
        button1.addActionListener(this);
        button2.addActionListener(this);
        button3.addActionListener(this);
    }

    @Override
    public void actionPerformed(ActionEvent ae) {
        String input=ae.getActionCommand();
        //System.out.println(input);
        if("Edit".equals(input)){
           frame1.setVisible(true);
           frame1.setSize(450,350);
           JPanel panel2=new JPanel();
           frame1.add(panel2);
           panel2.add(label1);
           panel2.add(field1);
           panel2.add(label2);
           panel2.add(field2);
           panel2.add(label3);
           panel2.add(field3);
           panel2.add(button3);
           Scanner sc=new Scanner(data);
            String[] item=new String[3];
                    int i=0;
                    while(sc.hasNext()){
                        item[i]=sc.next();
                        i++;
                    }
            for(int j=0;j<3;j++){
                if(j==0){
                    field1.setText(item[j]);
                }
                else if(j==1){
                    field2.setText(item[j]);
                }
                else{
                    field3.setText(item[j]);
                }
            }
            
            dispose();
        }else if("Save".equals(input)){  
            copy=field1.getText()+" "+field2.getText()+" "+field3.getText();
            try {
                Scanner sc = new Scanner(new File(name));
      StringBuffer buffer = new StringBuffer();
      while (sc.hasNextLine()) {
         buffer.append(sc.nextLine()+System.lineSeparator());
      }
      String fileContents = buffer.toString();
      sc.close();
      String oldLine = data;
      String newLine = copy;
      //Replacing the old line with new line
      fileContents = fileContents.replaceAll(oldLine, newLine);
      FileWriter writer = new FileWriter(name);
      writer.append(fileContents);
      writer.flush();
        } catch (IOException e) {
            e.printStackTrace();
        }
            frame1.dispose();
        }else if("Delete".equals(input)){
            copy="";
            try {
                Scanner sc = new Scanner(new File(name));
      StringBuffer buffer = new StringBuffer();
      while (sc.hasNextLine()) {
         buffer.append(sc.nextLine()+System.lineSeparator());
      }
      String fileContents = buffer.toString();
      sc.close();
      String oldLine = data;
      String newLine = copy;
      //Replacing the old line with new line
      fileContents = fileContents.replaceAll(oldLine, newLine);
      FileWriter writer = new FileWriter(name);
      writer.append(fileContents);
      writer.flush();
        } catch (IOException e) {
            e.printStackTrace();
        }
            frame1.dispose();
            dispose();
        }
    }
    
}


src/phonebook/Show.java

package phonebook;

import java.awt.Font;
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;
import javax.swing.*;


public class Show extends JFrame{
    JPanel panel=new JPanel();
    JLabel label[]=new JLabel[500];
    Show(){
        super("All Contact");
        setSize(350,800);
        add(panel);
        panel.setVisible(true);
        try{
                File obj=new File("History.txt");
                Scanner sc=new Scanner(obj);
                int i=0;
                while(sc.hasNextLine()){
                    label[i]= new JLabel(sc.nextLine());
                    label[i].setFont(new Font("Times New Roman", 1, 18));
                    panel.add(label[i]);
                    i++;
                }
                sc.close();
                setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
            }catch(FileNotFoundException e){
                System.out.println("File not Found");
                System.out.println(e);
            }
        }
}


src/phonebook/Search.java

package phonebook;

import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;
import javax.swing.*;


public class Search extends JFrame implements ActionListener{
    JFrame frame=new JFrame("Search Result");
    JButton button1;
    JPanel panel=new JPanel();
    JLabel label1=new JLabel("Name or Number");
    JTextField field1=new JTextField(17);
    
    Search(){
        super("Searching");
        setSize(450,200);
        add(panel);
        
        JButton button1=new JButton("Search");
        button1.setBackground(Color.cyan);
        
        panel.add(label1);
        label1.setFont(new Font("Times New Roman", 1, 18));
        field1.setFont(new Font("Times New Roman", 1, 18));
        field1.setBounds(100, 100, 200, 30);
        panel.add(field1);
        panel.add(button1);
        
        button1.addActionListener(this);
        
        //buttond.addActionListener(this);
    }

    @Override
    public void actionPerformed(ActionEvent ae) {
        String input=ae.getActionCommand();
        //System.out.println(input);
        if("Search".equals(input)){
            try{
                File obj=new File("History.txt");
                Scanner sc=new Scanner(obj);
                while(sc.hasNextLine()){
                    String data=sc.nextLine();
                    Scanner sc2=new Scanner(data);
                    while(sc2.hasNext()){
                        String item=sc2.next();
                        if(field1.getText().equalsIgnoreCase(item)){
                            Edit ob=new Edit(data);
                            ob.setVisible(true);
                        }
                    }
                    sc2.close();
                }
                sc.close();
            }catch(FileNotFoundException e){
                System.out.println("File not Found");
                System.out.println(e);
            }
            dispose();
        }
    }
    
}


src/phonebook/New_Contact.java

package phonebook;

import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.io.*;
import javax.swing.*;


public class New_Contact extends JFrame implements ActionListener {
    JFrame frame=new JFrame("Sorry");
    
    JButton button1;
    JPanel panel=new JPanel();
    JLabel label1=new JLabel("First Name");
    JTextField field1=new JTextField(20);
    JLabel label2=new JLabel("Last Name");
    JTextField field2=new JTextField(20);
    JLabel label3=new JLabel("Contact Number");
    JTextField field3=new JTextField(17);
    
    New_Contact(){
        super("Add new contact");
        
        setSize(450,240);
        add(panel);
        
        label1.setFont(new Font("Times New Roman", 1, 18));
        field1.setFont(new Font("Times New Roman", 1, 18));
        panel.add(label1);
        panel.add(field1);
        
        label2.setFont(new Font("Times New Roman", 1, 18));
        field2.setFont(new Font("Times New Roman", 1, 18));
        panel.add(label2);
        panel.add(field2);
        
        label3.setFont(new Font("Times New Roman", 1, 18));
        field3.setFont(new Font("Times New Roman", 1, 18));
        panel.add(label3);
        panel.add(field3);
        
        button1=new JButton("Save");
        panel.add(button1);
       button1.setFont(new Font("Times New Roman", 1, 28));
       button1.setBackground(Color.cyan);
        button1.setFocusable(false);
        button1.setLocation(230,100);
        button1.setPreferredSize(new Dimension(150, 50));
       button1.addActionListener(this);
    }

    @Override
    public void actionPerformed(ActionEvent ae) {
        String input=button1.getText();
        //System.out.println(input);
        if("Save".equals(input)){
            try {
                FileWriter f = new FileWriter("History.txt",true);
                if((field3.getText().length())==11){
                    //System.out.println(field3.getText().length());
                    int number=Integer.parseInt(field3.getText());
                    if( String.valueOf(number).length()==10){
                        //System.out.println(String.valueOf(number).length());
                f.write(field1.getText()+" "+field2.getText()+" "+field3.getText()+"\n");
                f.close();
                }else{
                    frame.setVisible(true);
                    frame.setSize(400,120);
                    JLabel label=new JLabel("Not a Bangladeshi number");
                    label.setFont(new Font("Times New Roman", 1, 32));
                    frame.add(label);
                }
                } else{
                    frame.setVisible(true);
                    frame.setSize(400,120);
                    JLabel label=new JLabel("Not a Bangladeshi number");
                    label.setFont(new Font("Times New Roman", 1, 32));
                    frame.add(label);
                }
                
            } catch (IOException ex) {
                System.out.println(ex);
            }
    }
            dispose();
    }
}


src/phonebook/PhoneMemory.java

package phonebook;

import java.awt.Dimension;
import java.awt.Font;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import javax.swing.*;


public class PhoneMemory extends JFrame implements ActionListener{
    JPanel panel=new JPanel();
    JButton button1=new JButton("Add New Contact");
    JButton button2=new JButton("Search");
    JButton button3=new JButton("Show all contact");
    JButton button4=new JButton("Exit");
    PhoneMemory(){
        super("Phonebook");
        add(panel);
        
        button1.setBounds(100, 160, 200, 40);
        button1.setFocusable(false);
        button1.setLocation(230,100);
        button1.setFont(new Font("Times New Roman", 1, 18));
        button1.setPreferredSize(new Dimension(400, 50));
        
        button2.setBounds(100, 160, 200, 40);
        button2.setFocusable(false);
        button2.setLocation(230,100);
        button2.setFont(new Font("Times New Roman", 1, 18));
        button2.setPreferredSize(new Dimension(400, 50));
        
        button3.setBounds(100, 160, 200, 40);
        button3.setFocusable(false);
        button3.setLocation(230,100);
        button3.setFont(new Font("Times New Roman", 1, 18));
        button3.setPreferredSize(new Dimension(400, 50));
        
        button4.setBounds(100, 160, 200, 40);
        button4.setFocusable(false);
        button4.setLocation(230,100);
        button4.setFont(new Font("Times New Roman", 1, 18));
        button4.setPreferredSize(new Dimension(400, 50));
        
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setSize(500,600);

        panel.add(button1);
        panel.add(button2);
        panel.add(button3);
        panel.add(button4);

        
        button1.addActionListener(this);
        button2.addActionListener(this);
        button3.addActionListener(this);
        button4.addActionListener(this);
    }

    @Override
    public void actionPerformed(ActionEvent event) {
        String input=event.getActionCommand();
        //System.out.println(input);
        if("Add New Contact".equals(input)){
            //frame.dispose();
           New_Contact obj=new New_Contact();
           obj.setVisible(true);
        }
        else if("Search".equals(input)){
            Search obj=new Search();
           obj.setVisible(true);
        }
        else if("Show all contact".equals(input)){
            Show obj=new Show();
            obj.setVisible(true);
        }
        else if("Exit".equals(input)){
            dispose();
        }
    }
}
